services/src/tpu_utils/tpu_connection_manager.rs

```diff
@@ -156,13 +151,12 @@ impl ActiveConnection {
 
                     let task_counter = task_counter.clone();
                     let connection_pool = connection_pool.clone();
-
+                    let permit = task_counter.acquire_owned().await.expect("Should get permit");
```

Comment:
could you put a comment, what the expected behavior should be wrt backpressure? more transactions will be .recv'd and block the select-handler?
wouldn't it be better to keep the transactions in the channel?